### PR TITLE
fixing Set and CGI signatures

### PR DESCRIPTION
**Pull request by HoneyryderChuck**
* Set: set the same type of generic element as for arrays;
* CGI: added missing modules (Cookie, Util, Escape...)

### CODE REVIEW

**Comment by HoneyryderChuck** [comment]
@soutaro I see there's already an open PR for `Tempfile`. Is there a solution thought out for delegate classes? Both these and Struct seem to be unaddressed when it comes to dynamic attributes.

File: stdlib/json/0/json.rbs [hunk 1]
```diff
@@ -560,7 +560,7 @@ class Regexp
   def to_json: (?JSON::State state) -> String
 end
 
-class Set[A]
+class Set[unchecked out A]
```
Comment (soutaro) [inline stdlib/json/0/json.rbs:563]: Could you update the type of `json_create?`. It has been broken...

```
def self.json_create: [A] (Hash[String, String | Array[A]] object) -> Set[A]
```

And, the same for other parameterized classes. 🙏

File: stdlib/set/0/set.rbs [hunk 2]
```diff
@@ -40,7 +40,8 @@
 #
 #     - Akinori MUSHA <knu@iDaemons.org> (current maintainer)
 #
-class Set[A]
+class Set[unchecked out A]
```
Comment (soutaro) [inline stdlib/set/0/set.rbs:43]: 👍

File: stdlib/openssl/0/openssl.rbs [hunk 3]
```diff
@@ -1441,9 +1441,9 @@ module OpenSSL
   end
 
   class HMAC
-    def self.digest: (String algo, String key, String data) -> String
+    def self.digest: (String | Digest algo, String key, String data) -> String
```
Comment (soutaro) [inline stdlib/openssl/0/openssl.rbs:1444]: 👍

File: stdlib/cgi/0/core.rbs [hunk 4]
```diff
@@ -560,36 +490,523 @@ class CGI
...
   def stdinput: () -> ::IO
 
   # Synonym for $stdout.
   #
   def stdoutput: () -> ::IO
-end
 
-# String for carriage return
-CGI::CR: String
+  CR: String
```
Comment (soutaro) [inline stdlib/cgi/0/core.rbs:587]: I want these constant declarations have comments... Seems like `annotate-with-rdoc` is still broken.

**Review by soutaro — CHANGES_REQUESTED** [review]
@HoneyryderChuck Could you update the type of `json_create`s?

File: stdlib/cgi/0/core.rbs (hunk 4 above)
Comment (soutaro) [reply stdlib/cgi/0/core.rbs:587]: I hope https://github.com/ruby/rbs/pull/768 prints the docs.

File: stdlib/json/0/json.rbs (hunk 1 above)
Comment (HoneyryderChuck) [reply stdlib/json/0/json.rbs:563]: done

File: stdlib/cgi/0/core.rbs (hunk 4 above)
Comment (HoneyryderChuck) [reply stdlib/cgi/0/core.rbs:588]: done, it did 👍

**Comment by HoneyryderChuck** [comment]
> And we are waiting for #767 get merged, right?

I'd say #767 is also dependent on #765 . Nevertheless, I'd be ok with merging it, as it's an acceptable workaround (and I just checked that `sorbet` did the same).

However, it's a hack, as `File` is not the superclass of `Tempfile` (`Delegator` is), and this will include `File` class methods in `Tempfile`, which is wrong.

This type of more "dynamic-ish" typing will become more important to figure out as the stdlib signature set completes, as I think it's relevant for a few of them such as `delegate`, `struct` and `ostruct`  (and `method_missing` delegations, perhaps). I've done a "draft" in #765 on how it could look like internally, but I think that the top-level API should look different and have specific RBS grammar.

**Comment by HoneyryderChuck** [comment]
@soutaro I think this is ready to be merged.

On the proposal above ☝️ , do you have some thoughts?

**Comment by HoneyryderChuck** [comment]
@soutaro ping ☝️

**Comment by HoneyryderChuck** [comment]
@soutaro just rebased the latest changes, can you give it another try?

**Review by soutaro — APPROVED** [review]

**Comment by soutaro** [comment]
@HoneyryderChuck Done! Fixed some issues myself and pushed to your repo. Thanks! 🙏

**Pull request merged**